platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/properties/PropertiesLoader.java

```diff
@@ -0,0 +1,226 @@
+        Properties properties = new Properties();
+        try {
+            LOGGER.debug(
+                    "Attempting to load properties from {} with Spring PropertiesLoaderUtils with class loader.",
+                    propertiesFile);
+            if (classLoader != null) {
+                properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile, classLoader);
+            } else {
+                properties = PropertiesLoaderUtils.loadAllProperties(propertiesFile,
+                        PropertiesLoader.class.getClassLoader());
+            }
+        } catch (IOException e) {
+            LOGGER.debug("Unable to load properties using default Spring properties loader.", e);
+        }
+
+        return properties;
+    }
+
+    /**
+     * Try loading the properties directly from the file system. If the properties file has a
+     * fully-qualified absolute path (which is what the blueprint file should specify) then it can
+     * be loaded directly, using this method.
+     */
+    private static Properties attemptLoadWithFileSystem(String propertiesFile,
+            ClassLoader classLoader) {
+        LOGGER.debug("Attempting to load properties from file system: {}", propertiesFile);
+        File propFile = new File(propertiesFile);
+        Properties properties = new Properties();
+
+        if (propFile.isAbsolute()) {
+            LOGGER.debug("propertiesFile {} is absolute", propertiesFile);
+            propFile = new File(propertiesFile);
+        } else {
+            String karafHome = System.getProperty("karaf.home");
+            if (karafHome != null && !karafHome.isEmpty()) {
+                propFile = new File(karafHome, propertiesFile);
+            } else {
+                karafHome = System.getProperty("ddf.home");
+                if (karafHome != null && !karafHome.isEmpty()) {
+                    propFile = new File(karafHome, propertiesFile);
+                } else {
+                    propFile = new File(propertiesFile);
+                }
+            }
+        }
+
+        try (InputStreamReader reader = new InputStreamReader(new FileInputStream(propFile),
+                StandardCharsets.UTF_8)) {
+            properties.load(reader);
+        } catch (FileNotFoundException e) {
+            LOGGER.debug("Could not find properties file: {}", propFile.getAbsolutePath(), e);
+        } catch (IOException e) {
+            LOGGER.debug("Error reading properties file: {}", propFile.getAbsolutePath(), e);
+            properties.clear();
+        }
+
+        return properties;
+    }
+
+    /**
+     * Try loading the properties using Java's resource loading facilities.
+     */
+    private static Properties attemptLoadAsResource(String propertiesFile,
+            ClassLoader classLoader) {
+        LOGGER.debug("Attempting to load properties as a resource: {}", propertiesFile);
+        InputStream ins = PropertiesLoader.class.getResourceAsStream(propertiesFile);
```

Comment:
Good question. Let's address this another time when we have a test harness to fall back on. I'm not quite comfortable changing this yet.